From 812adda1455b44082de5f599615d12bce906859a Mon Sep 17 00:00:00 2001
From: Jean-Christophe Morin <jcmorin@anaconda.com>
Date: Thu, 26 Oct 2023 10:37:14 -0400
Subject: [PATCH] Make FindCUDAToolkit required when building for GPU

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 718edeb..5ec2de3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,7 +140,7 @@ if (APPLE)
 endif()
 
 if (CT_CUBLAS)
-    find_package(CUDAToolkit)
+    find_package(CUDAToolkit REQUIRED)
     if (CUDAToolkit_FOUND)
         message(STATUS "cuBLAS found")
         enable_language(CUDA)
-- 
2.42.0

